crates/matrix-sdk/src/matrix_auth/mod.rs

```diff
@@ -292,6 +302,24 @@ impl MatrixAuth {
         LoginBuilder::new_token(self.clone(), token.to_owned())
     }
 
+    /// A higher level wrapper around the methods to complete an SSO login after
+    /// the user has logged in through a webview. This method should be used
+    /// in tandem with [`MatrixAuth::get_sso_login_url`].
+    pub fn login_with_sso_callback(&self, callback_url: Url) -> Result<LoginBuilder, SsoError> {
+        #[derive(Deserialize)]
+        struct QueryParameters {
+            #[serde(rename = "loginToken")]
+            login_token: Option<String>,
+        }
+
+        let query_string = callback_url.query().unwrap_or("");
```

Comment:
I think I would prefer `unwrap_or_default()` here, which avoid creating an empty `str` (the `Default` implementation always has an empty `str`, let's reuse this one instead).